src/main/java/it/gov/pagopa/arc/security/RedisOAuth2AuthorizationRequestRepository.java

```diff
@@ -38,7 +45,11 @@ public OAuth2AuthorizationRequest removeAuthorizationRequest(HttpServletRequest
     String state = request.getParameter(STATE);
     if (StringUtils.hasText(state)) {
       // Remove and return the authorization request from the map
-      return authorizationRequestMap.remove(state);
+      OAuth2AuthorizationRequest result = loadAuthorizationRequest(request);
```

Comment:
why don't you call directly the `oAuth2AuthorizationRequest.get(state)`

Reply:
ok